File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/GremlinFat_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";


export class GremlinFat_Model extends MonsterInfo{
    public static Id = "GremlinFat"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(14, 18);
        else
            return Mathf.RandomRange(13, 17);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/fatGremlin/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/fatGremlin/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/fatGremlin/skeleton.png"));
    }

    get Name(): string {
        return "胖地精";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/GremlinFat_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class GremlinFat_Model : MonsterInfo
    {
        public static string Id = "GremlinFat";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(14, 18) : Random.Range(13,17);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/fatGremlin/skeleton.atlas",
                    "Monsters/theBottom/fatGremlin/skeleton.json", "Monsters/theBottom/fatGremlin/skeleton.png");
            }
        }

        public override string Name => "胖地精";

        public GremlinFat_Model(Vector2 pos) : base(pos)
        {
        }
    }
}